calculator_test.go

```diff
@@ -2,23 +2,314 @@ package calculator_test
-	t.Parallel()
-	var want float64 = 4
-	got := calculator.Add(2, 2)
-	if want != got {
-		t.Errorf("want %f, got %f", want, got)
+	testCases := []struct {
+		name  string
+		want  float64
+		a     float64
+		b     float64
+		extra []float64
+	}{
+		{
+			name: "Add two numbers",
+			want: 4,
+			a:    2,
+			b:    2,
+		},
+		{
+			name:  "Sum four numbers",
+			want:  40,
+			a:     2,
+			b:     8,
+			extra: []float64{20, 10},
+		},
+		{
+			name: "Sum negative numbers",
+			want: -32,
+			a:    -16,
+			b:    -16,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Add(tC.a, tC.b, tC.extra...)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
+	}
+}
+
+func TestAddRandom(t *testing.T) {
+	var totalRandom int = 100
+	t.Run("Random number addition", func(t *testing.T) {
+		for count := 0; count < totalRandom; count++ {
+			var a float64 = rand.NormFloat64()*1024 + 2
+			var b float64 = rand.NormFloat64()*2048 + 5
+			want := a + b
+			got := calculator.Add(a, b)
+			if want != got {
+				t.Errorf("want %f, got %f", want, got)
+			}
+		}
+	})
+}
+
+func TestSubtract(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		a     float64
+		b     float64
+		extra []float64
+	}{
+		{
+			name: "Subtract a > b returns positive",
+			a:    3,
+			b:    2,
+			want: 1,
+		},
+		{
+			name: "Subtract a < b returns negative",
+			a:    3,
+			b:    5,
+			want: -2,
+		},
+		{
+			name: "Ensure decimals",
+			a:    6,
+			b:    2.225,
+			want: 3.77500,
+		},
+		{
+			name: "Subtract a and b negatives which returns a positive",
+			a:    -3,
+			b:    -5,
+			want: 2,
+		},
+		{
+			name:  "Subtract five numbers",
+			a:     20,
+			b:     5,
+			extra: []float64{4, 3, 7},
+			want:  1,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Subtract(tC.a, tC.b, tC.extra...)
+			if tC.want != got {
+				t.Errorf("want %f, got %f", tC.want, got)
+			}
+		})
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	testCases := []struct {
+		name  string
+		want  float64
+		a     float64
+		b     float64
+		extra []float64
+	}{
+		{
+			name: "Multiply two positive numbers returns positive",
+			a:    3,
+			b:    2,
+			want: 6,
+		},
+		{
+			name: "Multiply positive and negative should return negative",
+			a:    3,
+			b:    -2,
+			want: -6,
+		},
+		{
+			name: "Multiply two negative numbers returns positive",
+			a:    -3,
+			b:    -2,
+			want: 6,
+		},
+		{
+			name: "Multiply fractions return fraction",
+			a:    3.2,
+			b:    2.501,
+			want: 8.0032,
+		},
+		{
+			name: "Multiply by zero return zero",
+			a:    1943,
+			b:    0,
+			want: 0,
+		},
+		{
+			name:  "Multiply five numbers",
+			a:     2,
+			b:     2,
+			extra: []float64{5, 3, 2},
+			want:  120,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			got := calculator.Multiply(tC.a, tC.b, tC.extra...)
```

Comment:
You might find that you need to use a `closeEnough` function which determines if the difference is within some small value, depending on your test cases. Floating-point arithmetic is inherently imprecise; comparisons with `==` may well give surprising results.

Reply:
noted